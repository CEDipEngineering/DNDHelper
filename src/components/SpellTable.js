import { red } from '@material-ui/core/colors';
import React, { Component , Fragment} from 'react'
import { Input, Button } from "reactstrap";

var counter = 10;
class SpellTable extends React.Component {
    constructor(props){
        super()
        this.state = {
            loading: true,
            counter: counter,
        }
    }
   
    componentDidMount(){
        this.setState({loading: false, counter: counter})
    }
    render(){
        //console.log(this.props)
        var spellsArray = this.props.spellInfo;
        var rows = [];
        var nameList = new Array();
        for (let i =0; i<spellsArray.length; i++) {
            this.props.classes_filter.map(class_filter => {
                this.props.schools_filter.map(school_filter => {
                    this.props.levels_filter.map(level_filter => {
                        if (spellsArray[i].level.includes(level_filter.level) && spellsArray[i].school.includes(school_filter.school) && spellsArray[i].dnd_class.includes(class_filter.class) && class_filter.checked && school_filter.checked && level_filter.checked) {
                            if(nameList.includes(spellsArray[i].name)) {
                            } else {
                                nameList.push(spellsArray[i].name)
                                rows.push(
                                <tr>
                                    <td><a href={"https://open5e.com/spells/" + spellsArray[i].slug}>{spellsArray[i].name}</a></td>
                                    <td>{spellsArray[i].school}</td>
                                    <td>{spellsArray[i].level}</td>
                                    <td>{spellsArray[i].components}</td>
                                    <td>{spellsArray[i].dnd_class}</td>
                                </tr>
                                )
                            }
                            
                        }
                    })
                }) 
            })
        }
        return (
            
            <tr>
                {this.state.loading ? <p>carregando</p> : 
                    <div>
                        <thead>
                            <tr>
                                <th><Input type="text" onChange={this.props.allCallbacks.filter.func} value={this.props.allCallbacks.filter.state} /></th>
                                <th><Button id="filter" onClick={this.props.allCallbacks.searchSpell}>Search</Button></th>
                            </tr>
                            <th colspan="4" ><Button style = {{backgroundColor: "#808080",color : "#FFFFFF"}} outline value="prev" onClick={this.props.allCallbacks.previous}>Previous</Button></th>
                            <th><Button style = {{backgroundColor: "#808080",color : "#FFFFFF"}} outline value="next" onClick={this.props.allCallbacks.next}>Next</Button></th>
                            <tr>
                                <th><Button outline value="name" onClick={this.props.allCallbacks.check}>Name</Button></th>
                                <th><Button outline value="school" onClick={this.props.allCallbacks.check}>school</Button></th>
                                <th><Button outline value="level" onClick={this.props.allCallbacks.check}>Level</Button></th>
                                <th><Button outline value="component" onClick={this.props.allCallbacks.check}>Component</Button></th>
                                <th><Button outline value="class" onClick={this.props.allCallbacks.check}>Class</Button></th>
                            </tr>
                        </thead>
                        <tbody>               
                            {rows}
                        </tbody>
                        <th colspan="4" ><Button style = {{backgroundColor: "#808080",color : "#FFFFFF"}} outline value="prev" onClick={this.props.allCallbacks.previous}>Previous</Button></th>
                        <th><Button style = {{backgroundColor: "#808080",color : "#FFFFFF"}} outline value="next" onClick={this.props.allCallbacks.next}>Next</Button></th>

                    </div>
                }
            </tr>
        )
    }
} 

export default (SpellTable)