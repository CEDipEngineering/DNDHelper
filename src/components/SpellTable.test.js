import React from "react";
import { shallow } from "enzyme";
import SpellTabel from "./SpellTable";
import { Input, Button } from "./SpellTable";
import Enzyme from "enzyme";
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from"enzyme-to-json";

describe('SpellTabel', () => {
    it ("Randering pageSpells with no Spells or Filter", () => {
        const props = {
            spellInfo: [],
            allCallbacks:{filter:[]}
        }

        const wrapper = shallow(
            <SpellTabel {...props}/>
          );
        expect(toJson(wrapper)).toMatchSnapshot()
    });

    it ("Randering pageSpells with one spell and no Filter", () => {
        const props = { 
            spellInfo: {archetype: "Druid: Swamp",
            casting_time: "1 action",
            circles: "Swamp",
            components: "V, S, M",
            concentration: "no",
            desc: "A shimmering green arrow streaks toward a target within range and bursts in a spray of acid. Make a ranged spell attack against the target. On a hit, the target takes 4d4 acid damage immediately and 2d4 acid damage at the end of its next turn. On a miss, the arrow splashes the target with acid for half as much of the initial damage and no damage at the end of its next turn.",
            dnd_class: "Druid, Wizard",
            document__license_url: "http://open5e.com/legal",
            document__slug: "wotc-srd",
            document__title: "Systems Reference Document",
            duration: "Instantaneous",
            higher_level: "When you cast this spell using a spell slot of 3rd level or higher, the damage (both initial and later) increases by 1d4 for each slot level above 2nd.",
            level: "2nd-level",
            level_int: 2,
            material: "Powdered rhubarb leaf and an adder's stomach.",
            name: "Acid Arrow",
            page: "phb 259",
            range: "90 feet",
            ritual: "no",
            school: "Evocation",
            slug: "acid-arrow"},

            allCallbacks:{filter:[]}}  

        const wrapper = shallow(
            <SpellTabel {...props}/>
            )
        expect(toJson(wrapper)).toMatchSnapshot()
    });

    if ("Randering page when clicking next button", () =>{

        const props = {
            spellInfo: {slug: "acid-arrow"},
            allCallbacks:{filter:[],
                          next:true}
        }     

        const wrapper = shallow(
            <SpellTabel {...props}/>
            )
        expect(toJson(wrapper)).toMatchSnapshot()
    });



    });