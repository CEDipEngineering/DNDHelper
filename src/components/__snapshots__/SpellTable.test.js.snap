// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpellTabel Randering pageSpells with no Spells or Filter 1`] = `
<tr>
  <div>
    <thead>
      <tr>
        <th>
          <Input
            type="text"
          />
        </th>
        <th>
          <Button
            color="secondary"
            id="filter"
            tag="button"
          >
            Search
          </Button>
        </th>
      </tr>
      <th
        colspan="4"
      >
        <Button
          color="secondary"
          outline={true}
          style={
            Object {
              "backgroundColor": "#808080",
              "color": "#FFFFFF",
            }
          }
          tag="button"
          value="prev"
        >
          Previous
        </Button>
      </th>
      <th>
        <Button
          color="secondary"
          outline={true}
          style={
            Object {
              "backgroundColor": "#808080",
              "color": "#FFFFFF",
            }
          }
          tag="button"
          value="next"
        >
          Next
        </Button>
      </th>
      <tr>
        <th>
          <Button
            color="secondary"
            outline={true}
            tag="button"
            value="name"
          >
            Name
          </Button>
        </th>
        <th>
          <Button
            color="secondary"
            outline={true}
            tag="button"
            value="school"
          >
            school
          </Button>
        </th>
        <th>
          <Button
            color="secondary"
            outline={true}
            tag="button"
            value="level"
          >
            Level
          </Button>
        </th>
        <th>
          <Button
            color="secondary"
            outline={true}
            tag="button"
            value="component"
          >
            Component
          </Button>
        </th>
        <th>
          <Button
            color="secondary"
            outline={true}
            tag="button"
            value="class"
          >
            Class
          </Button>
        </th>
      </tr>
    </thead>
    <tbody />
    <th
      colspan="4"
    >
      <Button
        color="secondary"
        outline={true}
        style={
          Object {
            "backgroundColor": "#808080",
            "color": "#FFFFFF",
          }
        }
        tag="button"
        value="prev"
      >
        Previous
      </Button>
    </th>
    <th>
      <Button
        color="secondary"
        outline={true}
        style={
          Object {
            "backgroundColor": "#808080",
            "color": "#FFFFFF",
          }
        }
        tag="button"
        value="next"
      >
        Next
      </Button>
    </th>
  </div>
</tr>
`;

exports[`SpellTabel Randering pageSpells with one spell and no Filter 1`] = `
<tr>
  <div>
    <thead>
      <tr>
        <th>
          <Input
            type="text"
          />
        </th>
        <th>
          <Button
            color="secondary"
            id="filter"
            tag="button"
          >
            Search
          </Button>
        </th>
      </tr>
      <th
        colspan="4"
      >
        <Button
          color="secondary"
          outline={true}
          style={
            Object {
              "backgroundColor": "#808080",
              "color": "#FFFFFF",
            }
          }
          tag="button"
          value="prev"
        >
          Previous
        </Button>
      </th>
      <th>
        <Button
          color="secondary"
          outline={true}
          style={
            Object {
              "backgroundColor": "#808080",
              "color": "#FFFFFF",
            }
          }
          tag="button"
          value="next"
        >
          Next
        </Button>
      </th>
      <tr>
        <th>
          <Button
            color="secondary"
            outline={true}
            tag="button"
            value="name"
          >
            Name
          </Button>
        </th>
        <th>
          <Button
            color="secondary"
            outline={true}
            tag="button"
            value="school"
          >
            school
          </Button>
        </th>
        <th>
          <Button
            color="secondary"
            outline={true}
            tag="button"
            value="level"
          >
            Level
          </Button>
        </th>
        <th>
          <Button
            color="secondary"
            outline={true}
            tag="button"
            value="component"
          >
            Component
          </Button>
        </th>
        <th>
          <Button
            color="secondary"
            outline={true}
            tag="button"
            value="class"
          >
            Class
          </Button>
        </th>
      </tr>
    </thead>
    <tbody />
    <th
      colspan="4"
    >
      <Button
        color="secondary"
        outline={true}
        style={
          Object {
            "backgroundColor": "#808080",
            "color": "#FFFFFF",
          }
        }
        tag="button"
        value="prev"
      >
        Previous
      </Button>
    </th>
    <th>
      <Button
        color="secondary"
        outline={true}
        style={
          Object {
            "backgroundColor": "#808080",
            "color": "#FFFFFF",
          }
        }
        tag="button"
        value="next"
      >
        Next
      </Button>
    </th>
  </div>
</tr>
`;

exports[`SpellTabel Randering with no spells 1`] = `
<tr>
  <div>
    <thead>
      <tr>
        <th>
          <Input
            type="text"
          />
        </th>
        <th>
          <Button
            color="secondary"
            id="filter"
            tag="button"
          >
            Search
          </Button>
        </th>
      </tr>
      <th
        colspan="4"
      >
        <Button
          color="secondary"
          outline={true}
          style={
            Object {
              "backgroundColor": "#808080",
              "color": "#FFFFFF",
            }
          }
          tag="button"
          value="prev"
        >
          Previous
        </Button>
      </th>
      <th>
        <Button
          color="secondary"
          outline={true}
          style={
            Object {
              "backgroundColor": "#808080",
              "color": "#FFFFFF",
            }
          }
          tag="button"
          value="next"
        >
          Next
        </Button>
      </th>
      <tr>
        <th>
          <Button
            color="secondary"
            outline={true}
            tag="button"
            value="name"
          >
            Name
          </Button>
        </th>
        <th>
          <Button
            color="secondary"
            outline={true}
            tag="button"
            value="school"
          >
            school
          </Button>
        </th>
        <th>
          <Button
            color="secondary"
            outline={true}
            tag="button"
            value="level"
          >
            Level
          </Button>
        </th>
        <th>
          <Button
            color="secondary"
            outline={true}
            tag="button"
            value="component"
          >
            Component
          </Button>
        </th>
        <th>
          <Button
            color="secondary"
            outline={true}
            tag="button"
            value="class"
          >
            Class
          </Button>
        </th>
      </tr>
    </thead>
    <tbody />
    <th
      colspan="4"
    >
      <Button
        color="secondary"
        outline={true}
        style={
          Object {
            "backgroundColor": "#808080",
            "color": "#FFFFFF",
          }
        }
        tag="button"
        value="prev"
      >
        Previous
      </Button>
    </th>
    <th>
      <Button
        color="secondary"
        outline={true}
        style={
          Object {
            "backgroundColor": "#808080",
            "color": "#FFFFFF",
          }
        }
        tag="button"
        value="next"
      >
        Next
      </Button>
    </th>
  </div>
</tr>
`;
